src/openrct2/ride/Ride.cpp

```diff
@@ -6221,16 +6225,20 @@ CoordsXYZD ride_get_entrance_or_exit_position_from_screen_position(ScreenCoordsX
     word_F4418C = mapX;
     word_F4418E = mapY;
 
-    entranceExitCoords = { floor2(mapX, 32), floor2(mapY, 32), entranceExitCoords.z, INVALID_DIRECTION };
+    entranceExitCoords = { floor2(mapX, 32), floor2(mapY, 32), entranceExitCoords.z, stationHeight };
```

Comment:
Didn't realise station height was a type of direction 😜

Reply:
Wow, my bad, thanks for the catch! I've pushed the fix now